source/simulator/core.cc

```diff
@@ -1426,10 +1426,11 @@ namespace aspect
               build_advection_preconditioner(AdvectionField::composition(c),
                                              C_preconditioner);
               solve_advection(AdvectionField::composition(c));
-              current_linearization_point.block(introspection.block_indices.compositional_fields[c])
-                = solution.block(introspection.block_indices.compositional_fields[c]);
             }
 
+          for (unsigned int c=0; c<parameters.n_compositional_fields; ++c)
+            current_linearization_point.block(introspection.block_indices.compositional_fields[c])
+              = solution.block(introspection.block_indices.compositional_fields[c]);
```

Comment:
I had to think about this change for a while. It means that before, composition 2 already got to see the updated values for composition 1 that were computed previously. Now, that is no longer the case: they only get the composition 1 values from the previous time step.

I can see advantages to both schemes. Care to explain your reasoning?